packages/kbn-test/src/es/test_es_cluster.ts

```diff
@@ -19,10 +19,52 @@ import { esTestConfig } from './es_test_config';
 
 import { KIBANA_ROOT } from '../';
 
-interface TestClusterFactoryOptions {
+interface TestEsClusterNodesOptions {
+  name: string;
+  /**
+   * Depending on the test you are running, it may be necessary to
+   * configure a separate data archive for each node in the cluster.
+   * In that case, you can configure each of the archive paths here.
+   *
+   * Specifying a top-level `dataArchive` is not necessary if you are using
+   * this option; per-node archives will always be used if provided.
+   */
+  dataArchive?: string;
+}
+
+interface Node {
+  installSource: (opts: Record<string, unknown>) => Promise<{ installPath: string }>;
+  installSnapshot: (opts: Record<string, unknown>) => Promise<{ installPath: string }>;
+  extractDataDirectory: (
+    installPath: string,
+    archivePath: string,
+    extractDirName?: string
+  ) => Promise<{ insallPath: string }>;
+  start: (installPath: string, opts: Record<string, unknown>) => Promise<void>;
+  stop: () => Promise<void>;
+}
+
+export interface EsTestCluster {
+  ports: number[];
+  nodes: Node[];
+  getStartTimeout: () => number;
+  start: () => Promise<void>;
+  stop: () => Promise<void>;
+  cleanup: () => Promise<void>;
+  getClient: () => KibanaClient;
+  getUrl: () => string;
+  getHostUrls: () => string[];
+}
+
+export interface CreateTestEsClusterOptions {
+  /**
+   * Port to run Elasticsearch on. If you configure a
+   * multi-node cluster with the `nodes` option, this
+   * port will be incremented by one for each added node.
+   */
   port?: number;
   password?: string;
-  license?: 'basic' | 'trial'; // | 'oss'
+  license?: 'basic' | 'gold' | 'trial'; // | 'oss'
```

Comment:
Are we adding 'gold' license support for teams to use a multi-node setup for features that aren't available under the default distribution but they don't need all the functionality offered with a trial license?